src/_guides/language/effective-dart/usage.md

```diff
@@ -245,87 +245,66 @@ void error([String? message = null]) {
+against a Boolean literal is redundant. 
+It's always simpler to eliminate the equality operator, and add `!` if necessary:
 
 {:.good}
-<?code-excerpt "usage_good.dart (convert-null-aware)"?>
 {% prettify dart tag=pre+code %}
-// If you want null to be false:
-if (optionalThing?.isEnabled ?? false) {
-  print('Have enabled thing.');
-}
-
-// If you want null to be true:
-if (optionalThing?.isEnabled ?? true) {
-  print('Have enabled thing or nothing.');
-}
+if (nonNullableBool) { ... }
 {% endprettify %}
 
 {:.bad}
-<?code-excerpt "usage_bad.dart (convert-null-equals)"?>
 {% prettify dart tag=pre+code %}
-// If you want null to be false:
-if (optionalThing?.isEnabled == true) {
-  print('Have enabled thing.');
-}
-
-// If you want null to be true:
-if (optionalThing?.isEnabled != false) {
-  print('Have enabled thing or nothing.');
-}
+if (nonNullableBool == true) { ... }
 {% endprettify %}
 
-Both operations produce the same result and do the right thing, but `??` is
-preferred for three main reasons:
-
-*   The `??` operator signals that the code has something to do with `null`.
-
-*   The `== true` looks like a common mistake where the equality operator is
-    redundant and can be removed. That's true when the boolean expression on the
-    left will not produce `null`, but not when it can.
-
-*   The `?? false` and `?? true` clearly show what value will be used when the
-    expression is `null`. With `== true`, you have to think through the boolean
-    logic to realize that means that a `null` gets converted to *false*.
-
-**Exception:** Using a null-aware operator on a variable inside a condition
-doesn't promote the variable to a non-nullable type. If you want the variable
-to be promoted inside the body of the `if` statement, it might be better to use
-an explicit `!= null` check instead of `?.` followed by `??`:
+To evaluate a Boolean expression that *is nullable*, you should only use `??`
+or an explicit `!= null` check.
 
 {:.good}
-<?code-excerpt "usage_good.dart (null-aware-promote)"?>
 {% prettify dart tag=pre+code %}
-int measureMessage(String? message) {
-  if (message != null && message.isNotEmpty) {
-    // message is promoted to String.
-    return message.length;
-  }
+// If you want null to result in false:
+if (nullableBool ?? false) { ... }
 
-  return 0;
-}
+// If you want null to result in false
+// and you want the variable to type promote:
+if (nullableBool != null && nullableBool) { ... }
 {% endprettify %}
 
 {:.bad}
-<?code-excerpt "usage_bad.dart (null-aware-promote)"?>
 {% prettify dart tag=pre+code %}
-int measureMessage(String? message) {
-  if (message?.isNotEmpty ?? false) {
-    // message is not promoted to String.
-    return message!.length;
-  }
+// Error if null 
+if (nullableBool) { ... }
 
-  return 0;
-}
+// If you want null to be false:
+if (nullableBool == true) { ... }
 {% endprettify %}
 
+`if (nullableBool == true)` is a viable expression, but shouldn't be used for several reasons:
```

Comment:
I believe just the `nullableBool = true` portion of this is the expression. Maybe just stick to that as it should be avoided in other conditional situations as well (`nullableBool = true ? oneThing : SecondThing`):

```suggestion
`nullableBool == true` is a viable expression, 
but shouldn't be used for several reasons:
```